packages/supermassive/src/executeWithoutSchema.ts

```diff
@@ -961,60 +983,78 @@ function resolveAndCompleteField(
       hookContext = invokeBeforeFieldResolveHook(info, exeContext);
     }
 
-    const result = isPromise(hookContext)
-      ? hookContext.then((context) => {
-          hookContext = context;
-          return resolveFn(source, args, contextValue, info);
-        })
-      : resolveFn(source, args, contextValue, info);
+    let result: unknown;
+
+    if (hookContext instanceof GraphQLError) {
+      result = null;
+    } else if (isPromise(hookContext)) {
+      result = hookContext.then((context) => {
+        hookContext = context;
+
+        if (hookContext instanceof GraphQLError) {
+          return null;
+        }
+
+        return resolveFn(source, args, contextValue, info);
+      });
+    } else {
+      result = resolveFn(source, args, contextValue, info);
+    }
+
     let completed;
 
+    const handleAfterFieldHooks =
```

Comment:
Do we need a closure here or can this become a function?

Reply:
Agreed, this is a hot-path. It is too expensive to create a separate closure per every single field (especially if there is no actual hook defined)